docs/concepts/overview/extending.md

```diff
@@ -0,0 +1,209 @@
+---
+title: Extending your Kubernetes Cluster
+approvers:
+- erictune
+- lavalamp
+- cheftako
+- chenopis
+---
+
+{% capture overview %}
+
+Kubernetes has many ways to configure and extend it.  As a result,
+there is rarely a need to fork or submit patches to the Kubernetes
+project code.  
+
+This guide describes the options for customizing a kubernetes
+cluster.  It is aimed at **Cluster Administrators** who want to
+understand how to adapt their kubernetes cluster to the needs of
+their work environment.  Developers who are prospective **Extension
+Authors** and **Kubernetes Project Contributors** will also find it
+useful as an introduction to what extension points and patterns
+exist, and their tradeoffs and limitations.
+
+{% endcapture %}
+
+
+{% capture body %}
+
+## Overview
+
+Customization approaches can be broadly divided into **configuration**, which only involves changing flags, local configuration files, or API resources; and **extensions**, which involve running additional programs or services.  This document is primarily about extensions.
+
+## Configuration
+
+**Configuration files** and **flags** are documented in the Reference section of the online documentation, under each binary: [kubelet](/docs/admin/kubelet/), [kube-apiserver](/docs/admin/kube-apiserver/),  [kube-controller-manager](/docs/admin/kube-controller-manager/), and [kube-scheduler](/docs/admin/kube-scheduler/).  Flags and configuration files may not always be changable in a hosted Kubernetes service or a distribution with managed installation.  When they are changable, they are usually only changeable by the cluster administrator.    Also, they are subject to change in future Kubernetes versions, and setting them may require restarting processes.  For those reasons, they should be used only when there are no other options.
```

Comment:
Having the binaries as a bullet list will make for easier future adds/removes of items, versus a comma separated list.

Reply:
suggestion accepted.